PocketKit/Sources/PocketKit/MyList/SavedItemsList/SavedItemsListViewModel.swift

```diff
@@ -357,20 +371,31 @@ extension SavedItemsListViewModel {
         snapshot.reloadItems(cells)
         _snapshot = snapshot
     }
-    
+
+    private func applySorting(listOptions: ListOptions) {
+        let sortDescriptorTemp = NSSortDescriptor(key: "createdAt", ascending: (listOptions.selectedSort == .newest))
+        self.itemsController.sortDescriptors = [sortDescriptorTemp]
+        try? self.itemsController.performFetch()
+    }
+
     private func handleFilterSelection(with filter: ItemsListFilter) {
-        if filter == .all {
+        switch filter {
+        case .all:
             selectedFilters.removeAll()
             selectedFilters.insert(.all)
-        } else if selectedFilters.contains(filter) {
-            selectedFilters.remove(filter)
-        } else {
-            selectedFilters.insert(filter)
-            selectedFilters.remove(.all)
-        }
-        
-        if selectedFilters.isEmpty {
-            selectedFilters.insert(.all)
+        case .sortAndFilter:
+            presentedSortFilterViewModel = SortMenuViewModel(
+                source: Services.shared.source,
+                tracker: Services.shared.tracker.childTracker(hosting: .myList.myList),
+                listOptions: listOptions
```

Comment:
curious - could we use `source` and `tracker` passed in from the init?

Reply:
Seems to me, you are right @cyndichin , it's better to use source & tracker properties than shared but I might not have much clue, suggestions please.